src/USER-MISC/pair_wf_cut.cpp

```
@@ -0,0 +1,403 @@
+{
+  allocated = 1;
+  int n = atom->ntypes;
+
+  memory->create(setflag,n+1,n+1,"pair:setflag");
+  for (int i = 1; i <= n; i++)
+    for (int j = i; j <= n; j++)
+      setflag[i][j] = 0;
+
+  memory->create(cutsq,n+1,n+1,"pair:cutsq");
+
+  memory->create(cut,n+1,n+1,"pair:cut");
+  memory->create(epsilon,n+1,n+1,"pair:epsilon");
+  memory->create(sigma,n+1,n+1,"pair:sigma");
+  memory->create(nu,n+1,n+1,"pair:nu");
+  memory->create(mu,n+1,n+1,"pair:mu");
+  memory->create(nm,n+1,n+1,"pair:nm");
+  memory->create(e0nm,n+1,n+1,"pair:e0nm");
+  memory->create(rcmu,n+1,n+1,"pair:rcmu");
+  memory->create(sigma_mu,n+1,n+1,"pair:sigma_mu");
+  memory->create(offset,n+1,n+1,"pair:offset");
+}
+
+/* ----------------------------------------------------------------------
+   global settings
+------------------------------------------------------------------------- */
+
+void PairWFCut::settings(int narg, char **arg)
+{
+  if (narg != 1) error->all(FLERR,"Illegal pair_style command");
+
+  cut_global = utils::numeric(FLERR,arg[0],false,lmp);
+
+  // reset cutoffs that have been explicitly set
+
+  if (allocated) {
+    int i,j;
+    for (i = 1; i <= atom->ntypes; i++)
+      for (j = i; j <= atom->ntypes; j++)
+        if (setflag[i][j]) cut[i][j] = cut_global;
+  }
+}
+
+/* ----------------------------------------------------------------------
+   set coeffs for one or more type pairs
+------------------------------------------------------------------------- */
+
+void PairWFCut::coeff(int narg, char **arg)
+{
+  if (narg < 6 || narg > 7)
+    error->all(FLERR,"Incorrect args for pair coefficients");
+  if (!allocated) allocate();
+
+  int ilo,ihi,jlo,jhi;
+  utils::bounds(FLERR,arg[0],1,atom->ntypes,ilo,ihi,error);
+  utils::bounds(FLERR,arg[1],1,atom->ntypes,jlo,jhi,error);
+
+  double epsilon_one = utils::numeric(FLERR,arg[2],false,lmp);
+  double sigma_one = utils::numeric(FLERR,arg[3],false,lmp);
+  int nu_one = utils::numeric(FLERR,arg[4],false,lmp);
```

Since 'mu' and 'nu' are now integers, you should use `utils::inumeric()` instead of `utils::numeric()` to avoid having incorrect floating point number input be silently converted/truncated to integers.